app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeNameViewModel.kt

```diff
@@ -23,15 +24,15 @@ class AddSafeNameViewModel
     private val tracker: Tracker
 ) : BaseStateViewModel<BaseStateViewModel.State>(appDispatchers) {
 
-    fun submitAddressAndName(address: Solidity.Address, localName: String) {
+    fun submitAddressAndName(address: Solidity.Address, localName: String, chain: Chain) {
         safeLaunch {
             localName.takeUnless { it.isBlank() } ?: run {
                 updateState { AddSafeNameState(ViewAction.ShowError(InvalidName())) }
                 return@safeLaunch
             }
             updateState { AddSafeNameState(ViewAction.Loading(true)) }
             runCatching {
-                val safe = Safe(address, localName.trim())
+                val safe = Safe(address, localName.trim(), chain.chainId)
```

Comment:
I think we should also save chain data here. Currently it seems that chain is not saved to a db at all